src/common/adapter_manager.rs

```diff
@@ -47,50 +52,59 @@ where
     pub fn emit(&self, event: CentralEvent) {
         match event {
             CentralEvent::DeviceDisconnected(addr) => {
-                self.peripherals.remove(&addr);
+                self.shared.peripherals.remove(&addr);
             }
             CentralEvent::DeviceLost(addr) => {
-                self.peripherals.remove(&addr);
+                self.shared.peripherals.remove(&addr);
             }
             _ => {}
         }
 
-        send_notification(&self.async_senders, &event);
+        // Note: we ignore possible send errors which are expected while there
+        // could be zero recievers if application is not subscribed
+        let _ = self.shared.events_channel.send(event);
```

Comment:
Rather than ignoring the error entirely, how about logging it at trace level?

Reply:
okey, yip, I've pushed an update which traces any lost events due to having no subscribed receivers